timesketch/lib/tasks.py

```diff
@@ -71,97 +76,234 @@ def _set_timeline_status(index_name, status, error_msg=None):
+                    sketch_id, index_name, analyzer_name, **kwargs))
+        else:
+            tasks.append(run_sketch_analyzer.s(
+                sketch_id, index_name, analyzer_name))
+
+    return group(tasks)
+
+
+@celery.task(track_started=True)
+def run_index_analyzer(index_name, analyzer_name, **kwargs):
+    """Create a Celery task for an index analyzer.
+
+    Args:
+      index_name: Name of the datastore index.
+      analyzer_name: Name of the analyzer.
+
+    Returns:
+      Name (str) of the index.
+    """
+    analyzer_class = manager.AnalysisManager.get_analyzer(analyzer_name)
+    analyzer = analyzer_class(index_name=index_name, **kwargs)
+    result = analyzer.run_wrapper()
+    logging.info('[%s] result: %s' % (analyzer_name, result))
+    return index_name
+
+
+@celery.task(track_started=True)
+def run_sketch_analyzer(sketch_id, index_name, analyzer_name, **kwargs):
+    """Create a Celery task for a sketch analyzer.
+
+    Args:
+      sketch_id: ID of the sketch to analyze.
+      analyzer_name: Name of the analyzer.
+
+    Returns:
+      ID (str) of the sketch.
+    """
+    analyzer_class = manager.AnalysisManager.get_analyzer(analyzer_name)
+    analyzer = analyzer_class(
+        sketch_id=sketch_id, index_name=index_name, **kwargs)
+    result = analyzer.run_wrapper()
+    logging.info('[%s] result: %s' % (analyzer_name, result))
+    return sketch_id
+
+
 @celery.task(track_started=True, base=SqlAlchemyTask)
-def run_plaso(source_file_path, timeline_name, index_name, source_type,
-              username=None):
+def run_plaso(source_file_path, timeline_name, index_name, source_type):
     """Create a Celery task for processing Plaso storage file.
 
     Args:
         source_file_path: Path to plaso storage file.
         timeline_name: Name of the Timesketch timeline.
         index_name: Name of the datastore index.
         source_type: Type of file, csv or jsonl.
-        username: Username of the user who will own the timeline.
 
     Returns:
         String with summary of processed events.
     """
     # Log information to Celery
-    logging.info(u'Source type: %s', source_type)
+    message = 'Index timeline [{0:s}] to index [{1:s}] (source: {2:s})'
+    logging.info(message.format(timeline_name, index_name, source_type))
+
+    try:
+        psort_path = current_app.config['PSORT_PATH']
+    except KeyError:
+        psort_path = 'psort.py'
 
     cmd = [
-        u'psort.py', u'-o', u'timesketch', source_file_path, u'--name',
-        timeline_name, u'--status_view', u'none', u'--index', index_name
+        psort_path, '-o', 'timesketch', source_file_path, '--name',
+        timeline_name, '--status_view', 'none', '--index', index_name
     ]
 
-    if username:
-        cmd.append(u'--username')
-        cmd.append(username)
-
     # Run psort.py
     try:
-        cmd_output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+        subprocess.check_output(cmd, stderr=subprocess.STDOUT)
```

Comment:
Don't you need to add shell=True here in order for psort_path to work when it is set to just `psort.py`?

Reply:
Good point, but no, it seems to work without shell=True

Reply:
Interesting.  OK.